reposilite-site/docs/reverse-proxy-ssl.md

```diff
@@ -0,0 +1,86 @@
+---
+id: reverse-proxy-ssl
+title: Reverse proxy ssl
+sidebar_label: Reverse proxy ssl
+---
+
+Lots of people like to use a reverse proxy like nginx with reposilite. This is a page on
+how to setup Nginx with SSL.
+
+Source: <https://gist.github.com/Saladoc/8c17bda7b131c4fade9b2189ea5050d8>
+
+## Step 1
+First, install and setup Reposilite. Make sure to setup Reposilite to listen on port 8080 (or anything other than 80 and 443).
+
+Then, install nginx, openssl, and certbot [using snapd](https://snapcraft.io/docs/installing-snapd/) `sudo snap install certbot --classic` and `sudo ln -s /snap/bin/certbot /usr/bin/certbot`
+
+## Step 2
+Next you have to generate your certificates. To do this you will need a valid domain name and have your server pointed at it. 
+Run `sudo certbot certonly --standalone` and follow the instructions.
+Then, run `sudo mkdir /etc/nginx/ssl` and `sudo openssl dhparam -out /etc/nginx/ssl/dhparam.pem 2048` (also make sure www-data can read the file) This will take a while.
+
+## Step 3
+Create these files (replace repo.example.com with your domain):
+
+/etc/nginx/sites-available/reposilite-proxy.conf
+```conf
+# Prepare easy to use header value for websocket connections - needs to be outside server block
+
+map $http_upgrade $connection_upgrade {
+        default upgrade;
+        '' close;
+}
+
+server {
+
+        server_name repo.example.com;
+
+        listen 443 ssl http2;
+        listen [::]:443 ssl http2;
+
+        include /etc/nginx/custom-snippets/ssl.conf;
+
+        location / {
+                proxy_pass http://localhost:8080/;  # 8080 is the port reposilite is running on (change this to your port)
+                proxy_http_version 1.1;
+                proxy_set_header Upgrade $http_upgrade;
+                proxy_set_header Connection $connection_upgrade;
+                proxy_set_header Host $host;
+        }
+
+    ssl_certificate /etc/letsencrypt/live/repo.example.com/fullchain.pem; # managed by Certbot
+    ssl_certificate_key /etc/letsencrypt/live/repo.example.com/privkey.pem; # managed by Certbot
+}
+
+# Redirect all http requests to https
+
+server {
+        listen 80 default_server;
+        listen [::]:80 default_server;
+        return 301 https://$host$request_uri;
+}
+```
+
+/etc/nginx/custom-snippets/ssl.conf
```

Comment:
Maybe add a hint here that the contents of this file can also be inlined in place of the include directive, but it's handy to keep them in a separate file so that it's reusable. Likewise, `/etc/nginx/custom-snippets` is not a standard directory but one I made on my system.

The ssl settings should also be periodically reviewed and updated when new cyphers are added and others are deprecated. These settings right now get you an A+ on Qualys SSL Server test, but of course that may change over time.

Reply:
I added the hint and I’ll try to keep the page updated to standards :D